compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeParamAnalyzer.java

```diff
@@ -686,18 +687,9 @@ private BType getMatchingBoundType(BType expType, SymbolEnv env, HashSet<BType>
                         symTable.mapType.tsymbol);
             case TypeTags.STREAM:
                 BType streamConstraint = getMatchingBoundType(((BStreamType) expType).constraint, env, resolvedTypes);
-                BType streamError = ((BStreamType) expType).error;
-                if (streamError.tag == TypeTags.UNION) {
-                    List<BType> errorTypes = types.getAllTypes(streamError);
-                    if (errorTypes.stream().anyMatch(e -> e.tag == TypeTags.NEVER)) {
-                        streamError = symTable.neverType;
-                    } else {
-                        streamError = getMatchingOptionalBoundType((BUnionType) streamError, env, resolvedTypes);
-                    }
-                } else if (streamError.tag == TypeTags.ERROR) {
-                    streamError = getMatchingBoundType(streamError, env, resolvedTypes);
-                } else {
-                    streamError = symTable.neverType;
+                BType streamError = getMatchingBoundType(((BStreamType) expType).error , env, resolvedTypes);
+                if (streamError.tag == TypeTags.NONE) {
```

Comment:
Shall we add a small comment mentioning that we are intentionally adding nil if the matching bound type is not resolved.